mbuild/compound.py

```diff
@@ -324,6 +338,52 @@ def particles_by_element(self, element):
             if particle.element == element:
                 yield particle
 
+    @property
+    def mass(self):
+        """Return the total mass of a compound.
+
+        If the compound contains children compouds, the total mass of all
+        children compounds is returned.
+        If the compound contains element information (Compound.element) then
+        the mass is inferred from the elemental mass.
+        If Compound.mass has been set explicitly, then it will override the
+        mass inferred from Compound.element.
+        If neither of a Compound's element or mass attributes have been set,
+        then a mass of zero is returned.
+        """
+        if self._mass is not None:
+            return self._mass
+        else:
+            return sum([self._particle_mass(p) for p in self.particles()])
+
+    def _particle_mass(self, particle):
+        if particle._mass:
+            return particle._mass
+        else:
+            if particle.element:
+                return particle.element.mass
+            else:
+                return 0
+
+    @mass.setter
+    def mass(self, value):
+        if self.children:
+            raise MBuildError(
+                "Cannot set the mass of a Compound containing "
+                "children compounds"
+            )
+        if isinstance(value, Sequence) and not isinstance(value, str):
```

Comment:
Are there use-cases where you have to set mass from a sequence of length one?

Reply:
There was some discussion about this in a now resolved conversation around April 27th. Looking at it again, I think we can take out the checking for a sequence.